elasticsearch_collectd.py

```diff
@@ -587,25 +945,26 @@ def init_stats():
     else:
         ES_INDEX_URL = "http://" + ES_HOST + ":" + \
                        str(ES_PORT) + "/" + ",".join(ES_INDEX) + "/_stats"
+
     # common thread pools for all ES versions
     thread_pools = ['generic', 'index', 'get', 'snapshot', 'bulk', 'warmer',
                     'flush', 'search', 'refresh']
 
     if ES_VERSION.startswith("2."):
         thread_pools.extend(['suggest', 'percolate', 'management', 'listener',
-                             'force_merge', 'fetch_shard_store',
-                             'fetch_shard_started'])
+                             'fetch_shard_store', 'fetch_shard_started'])
+    elif ES_VERSION.startswith("2.") and not ES_VERSION.startswith("2.0"):
+        thread_pools.extend(['force_merge'])
     else:
         thread_pools.extend(['merge', 'optimize'])
 
-    # add info on thread pools (applicable for all versions)
-    for pool in thread_pools:
-        for attr in ['threads', 'queue', 'active', 'largest']:
-            path = 'thread_pool.{0}.{1}'.format(pool, attr)
-            NODE_STATS_CUR[path] = Stat("gauge", 'nodes.%s.{0}'.format(path))
-        for attr in ['completed', 'rejected']:
-            path = 'thread_pool.{0}.{1}'.format(pool, attr)
-            NODE_STATS_CUR[path] = Stat("counter", 'nodes.%s.{0}'.format(path))
+    # Legacy support for old configurations without Thread Pools configuration
+    if len(CONFIGURED_THREAD_POOLS) <= 0:
+        THREAD_POOLS = CONFIGURED_THREAD_POOLS
```

Comment:
I will change this and cast it to an array.  
This was to address the corner case you pointed out earlier in your comments where a customer does not have thread pools explicitly configured in their conf file.  In this case the CONFIGURED_THREAD_POOLS list is not configured at this point, and I assume a default state in which all applicable thread pools are monitored.